lich.rbw

```diff
@@ -2282,21 +2288,29 @@ module Games
                   Lich.log "Unclosed component tag detected: #{$_SERVERSTRING_.inspect}"
                   $_SERVERSTRING_ = "<component id='room exits'>Obvious paths: <d>clockwise</d>, <d>widdershins</d>.<compass></compass></component>"
                   Lich.log "Unclosed component tag fixed to: #{$_SERVERSTRING_.inspect}"
+                  # retry
                 end
                 # This is an actual DR line "<compass></compass></component>\r\n" which happens when the above is sent... subbing it out since we fix the tag above.
                 if $_SERVERSTRING_ == "<compass></compass></component>\r\n"
-                  Lich.log "Extraneous closing tag detected: #{$_SERVERSTRING_.inspect}"
+                  Lich.log "Extraneous closed tag detected: #{$_SERVERSTRING_.inspect}"
                   $_SERVERSTRING_ = ""
-                  Lich.log "Extraneous closing tag fixed: #{$_SERVERSTRING_.inspect}"
-                end                
-          
-                ## Clear out superfluous tags
-                $_SERVERSTRING_ = $_SERVERSTRING_.gsub("<pushStream id=\"combat\" /><popStream id=\"combat\" />","")
-                $_SERVERSTRING_ = $_SERVERSTRING_.gsub("<popStream id=\"combat\" /><pushStream id=\"combat\" />","")
+                  Lich.log "Extraneous closed tag fixed: #{$_SERVERSTRING_.inspect}"
+                end
 
-                ## Fix combat wrapping components - Why, DR, Why?
-                $_SERVERSTRING_ = $_SERVERSTRING_.gsub("<pushStream id=\"combat\" /><component id=","<component id=")
-                # $_SERVERSTRING_ = $_SERVERSTRING_.gsub("<pushStream id=\"combat\" /><prompt ","<prompt ")
+                # "<component id='room objs'>  You also see a granite altar with several candles and a water jug on it, and a granite font.\r\n"
+                # "<component id='room extra'>Placed around the interior, you see: some furniture and other bits of interest.\r\n
+                # Followed by in a new line.
+                # "</component>\r\n"
+                if $_SERVERSTRING_ =~ /^<component id='room (?:objs|extra)'>[^<]*(?!<\/component>)\r\n/
+                  Lich.log "Open-ended room objects component id tag: #{$_SERVERSTRING_.inspect}"
+                  $_SERVERSTRING_.gsub!("\r\n", "</component>")
```

Comment:
Should the ending be preserved? "</component>\r\n"

Reply:
No, it's not in every line, and breaks nothing to remove it.